### PR TITLE
feat(operator): Add Authentication section

### PR DESCRIPTION
**Pull request by Pothulapati**

### CODE REVIEW

File: docs/managing-dragonfly/operator/authentication.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
+---
+sidebar_position: 2
+---
+
+# Dragonfly Instance Authentication
+
+This guide provides step-by-step instructions for setting up Dragonfly with authentication. Currently, Dragonfly supports two types of authentication:
+
+- [Password-based authentication](#password-based-authentication) through a secret
```
Comment (kostasrim) [inline docs/managing-dragonfly/operator/authentication.md:9]: What about ACL's? Also keep in mind that if:

`requirepass=mypass` && `ACL SETUSER default newpass`

Then `AUTH mypass` will fail while `AUTH newpass` will succeed

File: docs/managing-dragonfly/operator/authentication.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+```sh
+kubectl describe dragonflies.dragonflydb.io dragonfly-sample
+```
+
+### Connecting to Dragonfly With TLS
+
+You should be able to connect to the Dragonfly instance only if you have a client certificate signed by the client CA.
+
+```sh
+kubectl run -it --rm redis-cli --image=redis:7.0.10 --restart=Never --overrides='
```
[the remark this answers is not in the document]
Comment (Pothulapati) [reply docs/managing-dragonfly/operator/authentication.md:208]: Yep. Its an issue. Fixed it. This depends on https://github.com/dragonflydb/dragonfly-operator/pull/124

File: docs/managing-dragonfly/operator/authentication.md (hunk 1 above)
Comment (Pothulapati) [reply docs/managing-dragonfly/operator/authentication.md:9]: But it feels like a Dragonfly docs thing right? and we don;t do anything w.r.t auth in the Operator yet. Worth to mention when we add something there? wdyt?

File: docs/managing-dragonfly/operator/server-tls.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---
-sidebar_position: 2
+sidebar_position: 3
```
Comment (Abhra303) [inline docs/managing-dragonfly/operator/server-tls.md:2]: you need to change sidebar_position

**Review by Abhra303 — APPROVED** [review]
Tested and working. LGTM

**Pull request merged**